internal/command/views/test.go

```diff
@@ -570,3 +578,45 @@ func testStatus(status moduletest.Status) string {
 		panic("unrecognized status: " + status.String())
 	}
 }
+
+// SaveErroredTestStateFile is a helper function to invoked in DestorySummary
+// to store the state to errored_test.tfstate and handle associated diagnostics and errors with this operation
+func SaveErroredTestStateFile(state *states.State, op Operation) tfdiags.Diagnostics {
```

Comment:
It's technically not `SaveErroredTestStateFile`'s place to be inside `DestroySummary`.

1. `DestroySummary` is part of the view, and is only a view action. 
2. There's another usage of `DestroySummary`, at the start of `cleanup`, in cases of inconsistent state (only caused by a bug in OpenTofu). Not sure we'd like to save the state in this case

Personally I'd use `SaveErroredTestStateFile` outside of `DestroySummary`

Reply:
I think this is a fair point. This is what I have in mind:

- The check for remaining resources is done by `state.HasManagedResourceInstanceObjects()`[here](https://github.com/opentofu/opentofu/blob/96b7fce5d1da52563260f6c445be14bfa0303357/internal/command/views/test.go#L225).

- Let's leave DestroySummary to perform logging/view related functionalities, writing to the file can be done in [here](url) instead, after invoking `state.HasManagedResourceInstanceObjects()` and checking if there are remnant resources.

WDYT?

Reply:
Sounds good.